pkg/search/postgres/common.go

```diff
@@ -589,6 +592,55 @@ func RunGetManyQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.
 	return results, nil
 }
 
+// RunCursorQueryForSchema creates a cursor against the database
+func RunCursorQueryForSchema(ctx context.Context, schema *walker.Schema, q *v1.Query, db *pgxpool.Pool) (fetcher func(n int) ([][]byte, error), closer func(), err error) {
+	query, err := standardizeQueryAndPopulatePath(q, schema, GET)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "error creating query")
+	}
+	if query == nil {
+		return nil, nil, emptyQueryErr
+	}
+
+	queryStr := query.AsSQL()
+
+	tx, err := db.Begin(ctx)
+	if err != nil {
+		closer()
+		return nil, nil, errors.Wrap(err, "creating transaction")
+	}
+	closer = func() {
+		if err := tx.Commit(ctx); err != nil {
+			log.Errorf("error rolling back cursor transaction: %v", err)
+		}
+	}
+
+	cursor := random.GenerateString(16, random.CaseInsensitiveAlpha)
```

Comment:
Since `GenerateString` uses crypto/rand, it may fail. Do I see it correctly that in such situation the opened transaction will not be closed?

Reply:
It panics as that should never happen so the whole application will fail

Reply:
I guess it could happen if there is not enough entropy for generating the random value. The whole application will fail, right, but an orphaned connection will be left and connection leaking normally is a problem.

Reply:
Thanks!